src/interpreter.jl

```diff
@@ -91,8 +90,9 @@ function Compiler.inlining_policy(interp::CthulhuInterpreter)
 end
 
 function Compiler.finish!(interp::CthulhuInterpreter, caller::InferenceResult)
-    if isa(caller.src, OptimizationState)
-        opt = caller.src
+    src = caller.src
+    if isa(src, OptimizationState)
```

Comment:
Run into something like this the other day, any way we can teach the compiler this?

Reply:
https://github.com/JuliaLang/julia/pull/41199 for immutables, and EscapeAnalysis.jl + effect analysis for mutables.